[[ Merge 6.7.8 ]] 30-09-15

livecodesebastien opened the pull request:
Refactored changes:
- addition of `MCPlayer::setdontuseqt` in player-platform.cpp implementation
- addition of PlaybleDuration property for Android Native player
- Update to StringRef the Environment variable building update for iPhone (mbliphonedc.mm)

Conflicts:
    engine/src/mbliphonedc.mm
    engine/src/player-platform.cpp
    engine/src/stack2.cpp
    ide
    version

`ide` submodule and version will be updated in a subsequent commit

livecodesebastien committed:
[[ Bug 15255 ]] Make DirectX the defult for revvideograbber

livecodepanos committed:
[Bug 15925] Implemented "playableDuration" for android player

livecodepanos committed:
Removed empty lines

livecodepanos committed:
Updated dict entry of mobileControlGet

livecodepanos committed:
Fixed conflict in dict entry

livecodesebastien committed:
[[ Bug 15987 ]] Use [NSProcessInfo environment] to build environment variables on iOS

livecodeian committed:
[[ Bug 15971 ]] Don't modify gradient when creating combiner

livecodesebastien committed:
[[ BUg 15987 ]] Update loop of environment NSdictionary elements to Obj-C

livecodepanos committed:
[Bug 11709] Make sure the vscroll/hscroll is set

livecodepanos committed:
[Bug 11709] Added bugfix note

livecodesebastien committed:
Update ide submodule

livecodesebastien committed:
Merge pull request #2917 from livecodepanos/bugfix-11709

[Bug 11709] Make sure the vscroll/hscroll is set

livecodesebastien committed:
[[ Bug 15875 ]] Update Android minimum SDK for Android deployment

Several changes:
 * manifest value `configChanges` also has `screenSize` set if min API >= 13
   (to avoid app restart on screen rotation)

 * minimum Android SDK needed by users for Android deployment is now 15,
   in order to accomodate with the previous need

 * minimum Android version been raised to 10 (as API 9 is not really relevant, since
   device running API 9 can update to use API 10)

 * config.h has been updated to now use android-10 as a minimum version

 * INSTALL-android.mk has been updated to reflect the android version change, and the
   update of the Platform tools to 23.0.1

 * A function returning the required SDK for Android deployment has been added in the script-only stack
   revdeploylibraryandroid (see counterpart changes in the ide).

livecodesebastien committed:
Merge remote-tracking branch 'origin/release-6.7.7' into develop-6.7

livecodesebastien committed:
[[ Bug 11133 ]] Add noui option for uninstaller

Add the missing 'runFaceless' command in card UninstallerUI in installer.rev

livecodesebastien committed:
Merge pull request #2926 from livecodesebastien/merge-release-6.7.7

Merge release 6.7.7

techstrategies committed:
[[ Bug 15873 ]] Separate the application of stack iconic properties from the (un)iconify actions

techstrategies committed:
[[ Bug 15873 ]] Set the iconic state of the stack if the stack window has opened iconified

techstrategies committed:
[[ Bug 15873 ]] Add release note

livecodesebastien committed:
[[ Bug 11133 ]] make runFaceless uninstall use actionUninstall

Change the call in runFaceless in stack "UninstallUI" to use actionUninstall instead of
installerRunUninstall (that was not passing the superuser to the uninstalling child process)

Add uninstall success message

livecodesebastien committed:
Merge pull request #2925 from livecodesebastien/bugfix-15875

[[ Bug 15875 ]] Update Android minimum SDK for Android deployment

livecodepanos committed:
Merge branch 'develop-6.7' of https://github.com/runrev/livecode into bugfix-15925

livecodepanos committed:
Removed "iOS only" from playableDuration property

livecodepanos committed:
Make sure Qt is (weak) linked when setting player's dontuseqt to false

livecodesebastien committed:
Merge pull request #2935 from livecodesebastien/bugfix-11133

[[ Bugfix 11133 ]] Implement faceless uninstaller

livecodesebastien committed:
[[ Bug 16035 ]] Initialise properly libURL at installer startup

revLibURL, now scriptified, needs to be loaded using its 'revLoadLibrary' function.
Simply putting its script in the backscripts when the installer stack was starting up
was not making revLibURL initialise its custom properties, and fail to check for the updates.

livecodesebastien committed:
Merge pull request #2910 from livecodeian/bugfix-15971

[[ Bug 15971 ]] Don't modify gradient when creating combiner

livecodesebastien committed:
Merge pull request #2939 from livecodesebastien/bugfix-16035

[[ Bug 16035 ]] Initialise properly libURL at installer startup

techstrategies committed:
[[ Bug 15873 ]] Always open window restored unless iconic state is true

livecodesebastien committed:
[[ Bug 11133 ]] Update release notes for noui uninstaller

techstrategies committed:
[[ Bug 15873 ]] Don't restore maximised windows to their original size, show them in their maximised state.

livecodesebastien committed:
[[ Bug 11133 ]] Remove version reference for command-line uninstaller

Remove the explicit reference of LiveCode 6.7.8 and LiveCode 7.1.1 as being the first versions with the noui uninstaller

livecodesebastien committed:
[[ Bug 15255 ]] Update bugfix note, add feature note

livecodesebastien committed:
Merge pull request #2875 from livecodepanos/bugfix-15925

[Bug 15925] Implemented "playableDuration" for android player

livecodesebastien committed:
[[ Bug 15987 ]] Avoid to cleanup unallocated memory in MCIphonDoDidBecomeActive

livecodesebastien committed:
Merge remote-tracking branch 'origin/develop-6.7' into bugfix-15255

livecodesebastien committed:
[[ Bug 15255 ]] Fix typo in void REVVideoGrabber

qvideograbber was typed instead of gvideograbber

livecodesebastien committed:
Merge pull request #2932 from techstrategies/bugfix-15873

[[ Bug 15873 ]] Set the iconic state of the stack if the stack window has opened iconified

livecodesebastien committed:
[[ Bug 11133 ]] Fix sentence in command-line uninstallation, in setup.md

livecodesebastien committed:
Merge pull request #2904 from livecodesebastien/bugfix-15987

[[ Bug 15987 ]] Use [NSProcessInfo environment] to build environment …

livecodesebastien committed:
[[ Bug 11133 ]] Fix few other mistakes in setup.md

livecodesebastien committed:
Merge pull request #2212 from livecodesebastien/bugfix-15255

[[ Bug 15255 ]] Make DirectX the defult for revvideograbber

livecodesebastien committed:
[[ Bug 12876 ]] make the installer calls the right LiveCode exec name

The executable name still had its spaces removed, and was lowercased - which does
not comply with the new naming.

livecodesebastien committed:
Merge pull request #2943 from livecodesebastien/bugfix-11133_release_notes

[[ Bug 11133 ]] Update release notes for noui uninstaller

livecodesebastien committed:
Merge pull request #2937 from livecodepanos/bugfix-13820_revisited

Make sure QT is (weak) linked when setting player's dontuseqt to false

livecodesebastien committed:
[[ Bug 16047 ]] Make MCS_startprocess give the right args to execvp

The variable `name` was modified and no longer had the final quote of the executable name, once the `argv` list had been build.
Change the parameteres given to `execvp` to `argv[0], argv`.

livecodesebastien committed:
[[ Bug 12876 ]] Properly check the version being installed

livecodesebastien committed:
Merge pull request #2951 from livecodesebastien/bugfix-16047

[[ Bug 16047 ]] Make MCS_startprocess give the right args to execvp

livecodesebastien committed:
Merge pull request #2949 from livecodesebastien/bugfix-12876_7.0

[[ Bug 12876 ]] make the installer calls the right LiveCode exec name

livecodesebastien committed:
Merge remote-tracking branch 'origin/develop-6.7' into develop-7.0

Refactored changes:
  * addition of `MCPlayer::setdontuseqt` in player-platform.cpp implementation
  * addition of PlaybleDuration property for Android Native player
  * Update to StringRef the Environment variable building update for iPhone (mbliphonedc.mm)

Conflicts:
	engine/src/mbliphonedc.mm
	engine/src/player-platform.cpp
	engine/src/stack2.cpp
	ide

livecodesebastien committed:
[[ LC 6.7.8 RC 1 ]] Update ide submodule

livecodesebastien committed:
[[ LC 6.7.8. RC 1 ]] Bump version number

livecodesebastien committed:
Merge remote-tracking branch 'origin/develop-6.7' into develop-7.0

Conflicts solved by ignoring changes (ide submodule and version will be updated in a subsequent commit)

Conflicts:
	ide
	version

livecodeali commented:
@livecode-vulcan review ok 240bdd2

livecode-vulcan commented:
💙 review by @livecodeali ok 240bdd20ae82926bd5516882afa0ca5f47329702

livecode-vulcan commented:
😞 test failure 240bdd20ae82926bd5516882afa0ca5f47329702
- try-community-linux-x86: [pending](https://vulcan.livecode.com/builders/try-community-linux-x86/builds/545)
- try-community-mac: [failure](https://vulcan.livecode.com/builders/try-community-mac/builds/549)
- try-community-linux-x86_64: [pending](https://vulcan.livecode.com/builders/try-community-linux-x86_64/builds/544)
- try-community-emscripten: [success](https://vulcan.livecode.com/builders/try-community-emscripten/builds/544)
- try-community-ios: [failure](https://vulcan.livecode.com/builders/try-community-ios/builds/553)
- try-community-win-x86: [pending](https://vulcan.livecode.com/builders/try-community-win-x86/builds/547)
- try-community-android-armv6: [failure](https://vulcan.livecode.com/builders/try-community-android-armv6/builds/555)

livecodepanos commented on engine/src/player-platform.cpp:
```diff
@@ -1951,6 +1953,21 @@ void MCPlayer::setlooping(Boolean loop)
+	}
+}
+
 real8 MCPlayer::getplayrate()
```
I think the signature of setdontuseqt() has to be added in player-platform.h

livecodesebastien committed:
[[ Merge 6.7.8 RC 1 ]] Fix compile error on Mac and iOS

Add 'void setdontuseqt' in player-platform.h definition of MCPlayer

mbliphonedc.mm:
  * Revert change to call to X_init
  * Update type of t_index to avoid warnings

livecodeali commented:
@livecode-vulcan review ok 9616d95

livecode-vulcan commented:
💙 review by @livecodeali ok 9616d95366a0a4ef0ab514e56a3b8ed58c7395f9

livecode-vulcan commented:
😞 test failure 9616d95366a0a4ef0ab514e56a3b8ed58c7395f9
- try-community-linux-x86: [pending](https://vulcan.livecode.com/builders/try-community-linux-x86/builds/547)
- try-community-mac: [success](https://vulcan.livecode.com/builders/try-community-mac/builds/551)
- try-community-linux-x86_64: [pending](https://vulcan.livecode.com/builders/try-community-linux-x86_64/builds/546)
- try-community-emscripten: [success](https://vulcan.livecode.com/builders/try-community-emscripten/builds/546)
- try-community-ios: [pending](https://vulcan.livecode.com/builders/try-community-ios/builds/555)
- try-community-win-x86: [pending](https://vulcan.livecode.com/builders/try-community-win-x86/builds/549)
- try-community-android-armv6: [failure](https://vulcan.livecode.com/builders/try-community-android-armv6/builds/557)

livecodesebastien committed:
[[ Merge 6.7.8 RC 1 ]] Fix wrong variable naming in GetPlayableDuration

livecodeali commented:
@livecode-vulcan review ok 0b49cf8

livecode-vulcan commented:
💙 review by @livecodeali ok 0b49cf86a1bed5de13009902a4b5bd0fb3ddf3a0

livecode-vulcan commented:
😎 test success 0b49cf86a1bed5de13009902a4b5bd0fb3ddf3a0
- try-community-linux-x86: [success](https://vulcan.livecode.com/builders/try-community-linux-x86/builds/548)
- try-community-mac: [success](https://vulcan.livecode.com/builders/try-community-mac/builds/553)
- try-community-linux-x86_64: [success](https://vulcan.livecode.com/builders/try-community-linux-x86_64/builds/549)
- try-community-emscripten: [success](https://vulcan.livecode.com/builders/try-community-emscripten/builds/547)
- try-community-ios: [success](https://vulcan.livecode.com/builders/try-community-ios/builds/556)
- try-community-win-x86: [success](https://vulcan.livecode.com/builders/try-community-win-x86/builds/550)
- try-community-android-armv6: [success](https://vulcan.livecode.com/builders/try-community-android-armv6/builds/558)
